packages/test/test-end-to-end-tests/src/test/handleValidation.spec.ts

```diff
@@ -0,0 +1,425 @@
+import type { SharedString } from "@fluidframework/sequence/internal";
+import {
+	ChannelFactoryRegistry,
+	ITestContainerConfig,
+	DataObjectFactoryType,
+	createAndAttachContainer,
+	ITestFluidObject,
+	waitForContainerConnection,
+} from "@fluidframework/test-utils/internal";
+
+import type { ISharedMatrix } from "@fluidframework/matrix/internal";
+import type { IConsensusRegisterCollection } from "@fluidframework/register-collection/internal";
+import {
+	ConsensusResult,
+	type ConsensusCallback,
+	type ConsensusQueue,
+	type IConsensusOrderedCollection,
+} from "@fluidframework/ordered-collection/internal";
+import {
+	ISharedTree,
+	SharedTree,
+	SchemaFactory,
+	TreeConfiguration,
+} from "@fluidframework/tree/internal";
+
+const mapId = "map";
+const stringId = "sharedString";
+const cellId = "cell";
+const counterId = "counter";
+const directoryId = "directory";
+const treeId = "tree";
+const matrixId = "matrix";
+const legacyTreeId = "legacyTree";
+const registerId = "registerCollection";
+const queueId = "consensusQueue";
+const migrationShimId = "migrationShim";
+
+const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+	getRawConfig: (name: string): ConfigTypes => settings[name],
+});
+
+function treeSetup (dds) {
+	const builder = new SchemaFactory("test");
+				class Bar extends builder.object("bar", {
+					h: builder.optional(builder.handle),
+				}) {}
+
+				const config = new TreeConfiguration(Bar, () => ({
+					h: undefined,
+				}));
+
+				const treeView = dds.schematize(config);
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+				return treeView;
+}
+
+describeCompat("handle validation", "NoCompat", (getTestObjectProvider, apis) => {
+	const {
+		SharedMap,
+		SharedDirectory,
+		SharedCounter,
+		SharedString,
+		SharedCell,
+		SharedMatrix,
+		ConsensusRegisterCollection,
+		ConsensusQueue,
+	} = apis.dds;
+
+	const registry: ChannelFactoryRegistry = [
+		[mapId, SharedMap.getFactory()],
+		[stringId, SharedString.getFactory()],
+		[cellId, SharedCell.getFactory()],
+		[counterId, SharedCounter.getFactory()],
+		[directoryId, SharedDirectory.getFactory()],
+		[treeId, SharedTree.getFactory()],
+		[matrixId, SharedMatrix.getFactory()],
+		// [legacyTreeId, LegacySharedTree.getFactory()],
+		[registerId, ConsensusRegisterCollection.getFactory()],
+		[queueId, ConsensusQueue.getFactory()],
+		// [
+		// 	migrationShimId,
+		// 	new MigrationShimFactory(
+		// 		LegacySharedTree.getFactory(),
+		// 		SharedTree.getFactory(),
+		// 		(legacyTree, newTree) => {
+		// 			throw new Error("unreachable");
+		// 		},
+		// 	),
+		// ],
+	];
+
+	const testContainerConfig: ITestContainerConfig = {
+		fluidDataObjectType: DataObjectFactoryType.Test,
+		registry,
+		runtimeOptions: {
+			enableRuntimeIdCompressor: "on",
+		},
+		loaderProps: {
+			configProvider: configProvider({
+				"Fluid.Container.enableOfflineLoad": true,
+			}),
+		},
+	};
+
+	async function setup() {
```

Comment:
pull this function and the above consts out of the describeCompat block, they can just be above it in the file